pkg/icingadb/sla.go

```diff
@@ -0,0 +1,132 @@
+package icingadb
+
+import (
+	"context"
+	"github.com/icinga/icingadb/pkg/contracts"
+	v1 "github.com/icinga/icingadb/pkg/icingadb/v1"
+	"github.com/icinga/icingadb/pkg/types"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+	"time"
+)
+
+type SlaHistoryTrail struct {
+	v1.EntityWitNoneBinaryId `json:",inline"`
+	v1.EnvironmentMeta       `json:",inline"`
+	HostId                   types.Binary    `json:"host_id"`
+	ServiceId                types.Binary    `json:"service_id"`
+	EventType                string          `json:"event_type"`
+	EventTime                types.UnixMilli `json:"event_time"`
+}
+
+// ID implements part of the contracts.IDer interface.
+func (sht SlaHistoryTrail) ID() contracts.ID {
+	return sht.Id
+}
+
+// SetID implements part of the contracts.IDer interface.
+func (sht *SlaHistoryTrail) SetID(id contracts.ID) {
+	sht.Id = id.(types.Int)
+}
```

Comment:
Apropos the mentioned struct w/ autoinc. IDs: these methods could be part of it.